src/middleware/usbboot.rs

```diff
@@ -41,6 +41,36 @@ impl fmt::Display for FlashingError {
 
 impl std::error::Error for FlashingError {}
 
+pub trait FlashingErrorExt<T, E: Display> {
+    fn map_err_into_logged_usb(self, logging: &Sender<FlashProgress>) -> Result<T, FlashingError>;
+    fn map_err_into_logged_io(self, logging: &Sender<FlashProgress>) -> Result<T, FlashingError>;
+}
+
+impl<T, E: Display> FlashingErrorExt<T, E> for Result<T, E> {
+    fn map_err_into_logged_usb(self, logging: &Sender<FlashProgress>) -> Result<T, FlashingError> {
+        self.map_err(|e| {
+            logging
+                .try_send(FlashProgress {
```

Comment:
Personally, in addition to these 2 traits, I'd add easier-to-use `FlashProgress` methods to reduce boilerplate. For example, `FlashProgress::new(msg, status).send(logger)`, `FlashProgress::from_err(msg, err)`.

This is not a suggestion for this PR, just to consider for future development.

Reply:
Good suggestion! I had an idea to drop the FlashProgress all together. I wrote it down in #32